### PR TITLE
Add a simple /statusz page to help introspect running servers.

### PR DESCRIPTION
**Pull request by tylerwilliams**
This pretty much mimics a 'traditional' statusz endpoint and allows various backend services to report their own status in an easily digestible (html) format.

Screenshot, to convey the gist:

image

Fixes: https://github.com/buildbuddy-io/buildbuddy-internal/issues/139

### CODE REVIEW

File: server/util/statusz/statusz.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+	  float: left;
+	  width: 80%;
+	}
+	.righthand {
+	  text-align: right;
+	}
+    </style>
+  </head>
+    <h1>Status for {{.BinaryName}}</h1>
+  <div>
```
Comment (siggisim) [inline server/util/statusz/statusz.go:50]: as mentioned above, i'd apply a class to this div and (once it's closed - see below comment) apply `display: flex` to it

File: server/util/statusz/statusz.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  <div class=lefthand>
+    Started at <span>{{.StartTime.Format "Jan 02, 2006 15:04:05 PST" }}</span><br>
+    Current time <span>{{.CurrentTime.Format "Jan 02, 2006 15:04:05 PST" }}</span><br>
+    App Version <span>{{.AppVersion}} ({{.Commit}})</span><br>
+    Go Version <span>{{.GoVersion}}</span><br>
+  </div>
+  <div class=righthand>
```
Comment (siggisim) [inline server/util/statusz/statusz.go:57]: once you apply flexbox to the parent div, you can remove these `class=righthand` and `class=lefthand` classes.

if you didn't they should be quoted `class="righthand"`

**Review by siggisim — APPROVED** [review]
Looks good, just a lot of css & html nits.

File: server/util/statusz/statusz.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+type Section struct {
+	Name        string
+	Description string
+	Fn          StatusFunc
+}
+
+type Handler struct {
+	mu       *sync.RWMutex // PROTECTS(sections)
```
Comment (vadimberezniker) [inline server/util/statusz/statusz.go:106]: nit: does this need to be a pointer? this could simply be `mu sync.RWMutex`

**Review by vadimberezniker — APPROVED** [review]
LGTM

Looks good. Thanks for laying the foundation for exposing more debug data.

**Review by tylerwilliams — COMMENTED** [review]
PTAL

**Pull request merged**